ignite/contrib/handlers/clearml_logger.py

```diff
@@ -435,25 +478,29 @@ def __call__(self, engine: Engine, logger: ClearMLLogger, event_name: Union[str,
 
         global_step = engine.state.get_event_attrib_value(event_name)
         tag_prefix = f"{self.tag}/" if self.tag else ""
-        for name, p in self.model.named_parameters():
-            if p.grad is None:
-                continue
+        for name, p in self.weights:
 
             title_name, _, series_name = name.partition(".")
             logger.clearml_logger.report_scalar(
                 title=f"{tag_prefix}weights_{self.reduction.__name__}/{title_name}",
                 series=series_name,
-                value=self.reduction(p.data),
+                value=self.reduction(p.data.detach()).cpu(),
```

Comment:
Here you made this assumption on the output as tensor. This code will fail if output is python float and which is a possible reduction function output. I do not think that we want to change that.

Reply:
I changed `reduction` type hint to `Callable[[torch.Tensor], Any]`.

Reply:
How about replacing `Any` by `Union[Number, torch.Tensor]` ?

Reply:
Yes. That sounds better.

Reply:
We may fix mypy issue with visdom as : 
1) ignoring type mismatch
2) Use `Union[float, torch.Tensor]`
I let you decide

Reply:
The safer, the better.
I've got to change `_is_0D_tensor` parameter type hint as well. It wants to check if its argument is a tensor and beside that a 1D tensor so being `Any` is a reasonable assumption.